Iced/Intel/EncoderInternal/OpHandlers.cs

```diff
@@ -629,6 +635,26 @@ sealed class OpModRM_rm : Op {
 			encoder.AddRegOrMem(ref instr, operand, regLo, regHi, allowMemOp: true, allowRegOp: true);
 	}
 
+	sealed class OpRegOpCodeByte: Op {
+		readonly Register regLo;
+		readonly Register regHi;
+
+		public OpRegOpCodeByte(Register regLo, Register regHi) {
+			this.regLo = regLo;
+			this.regHi = regHi;
+		}
+
+		public override void Encode(Encoder encoder, ref Instruction instr, int operand) 
+		{
+			if (instr.GetOpKind(operand) == OpKind.Register) 
+			{
+				var register = instr.GetOpRegister(operand);
+				encoder.Verify(operand, register, regLo, regHi);
+				encoder.OpCode = encoder.OpCode + (uint)(register - regLo);
```

Comment:
This works if only low 8 registers are used. If this handler gets reused for all 16 regs, you'll need a `&7` here too

Reply:
Also if it's reg r8-r15, you need to set encoder.Flags' REX.W bit. There's also the 8bit regs that need special handling. The encoder already has some logic in `AddModRMRegister`, but it's for modrm instructions, you could add a new method.

Reply:
So just to make sure, I would add a new `OpRegEmbed32`  that would take two ranges of register RAX-RDI and R8W-R15W, and I would detect either R8W-R15W  usage to add proper REX bits, that's correct?

What about the name `OpRegOpCodeByte`, should I use something like  `OpRegEmbed8` instead?

Reply:
Ok, though for the following case:

```c#
				(uint)EncFlags1.Legacy | (uint)Code.Push_AX | ((uint)0x50 << (int)EncFlags1.OpCodeShift), (uint)LegacyFlags.Encodable_Any | (uint)LegacyFlags.o16, (uint)LegacyOpKind.AX,
				(uint)EncFlags1.Legacy | (uint)Code.Push_R8W | ((uint)0x50 << (int)EncFlags1.OpCodeShift), (uint)LegacyFlags.Encodable_Only64 | (uint)LegacyFlags.o16_rexb, (uint)LegacyOpKind.R8W,
				(uint)EncFlags1.Legacy | (uint)Code.Push_EAX | ((uint)0x50 << (int)EncFlags1.OpCodeShift), (uint)LegacyFlags.Encodable_Only1632 | (uint)LegacyFlags.o32, (uint)LegacyOpKind.EAX,
				(uint)EncFlags1.Legacy | (uint)Code.Push_RAX | ((uint)0x50 << (int)EncFlags1.OpCodeShift), (uint)LegacyFlags.Encodable_Only64, (uint)LegacyOpKind.RAX,
				(uint)EncFlags1.Legacy | (uint)Code.Push_R8 | ((uint)0x50 << (int)EncFlags1.OpCodeShift), (uint)LegacyFlags.Encodable_Only64 | (uint)LegacyFlags.rexb, (uint)LegacyOpKind.R8,
```

How can I put a single line for `Push_r16` while the Legacy flags can be `LegacyFlags.Encodable_Any | (uint)LegacyFlags.o16,` for AX-DI and `LegacyFlags.Encodable_Only64 | (uint)LegacyFlags.o16_rexb` for R8W-R15W?

Reply:
Or should I add for them a `Push_r16_rexb`?

Reply:
You can remove the rexb part. The handler should set it if it's a high reg (or better, a new Encoder method called by the handler that has all the logic). In other words, the lines should look very similar to the Inc_xxx lines. Encodable_Any should be used in this case though since the instructions are available in all modes and not only 16/32-bit modes (inc/dec r16/32)

> Or should I add for them a Push_r16_rexb?

No, Push_r16 can support all 16 regs.

Reply:
yep ok, will try to do that then